### PR TITLE
Improved output logic

### PR DESCRIPTION
**Pull request by tverbeiren**

### CODE REVIEW

File: src/runner/config.vsh.yaml [hunk 1]
```diff
@@ -4,7 +4,9 @@ argument_groups:
   - name: Input arguments
     arguments:
       - name: --input
-        description: Base directory of the form `s3:/<bucket>/Sequencing/<Sequencer>/<RunID>/`
+        description: |
+          Base directory of the canonical form `s3://<bucket>/<path>/<RunID>/`.
+          Please note that the workflow supports xxx
```
Comment (DriesSchaumont) [inline src/runner/config.vsh.yaml:9]: What does `xxx` mean here? 🤔

File: src/runner/main.nf [hunk 2]
```diff
@@ -9,6 +9,16 @@ workflow run_wf {
 
   main:
     output_ch = input_ch
+      // Extract the ID from the input.
+      // If this is a directory or tar-gz file containing `.`'s,
+      // only select the part before the first `.`.
+      | map{ id, state ->
+        def id_with_dots = state.input.getFileName().toString()
```
Comment (DriesSchaumont) [inline src/runner/main.nf:16]: In the actual workflow we detect a tar file with `.tar.gz`, `.tar` and `.tgz`. I think we could just strip those three here and use that? This avoids problems with extra dots like `this.is.an.example.tar.gz`

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]

File: src/runner/main.nf [hunk 3]
```diff
@@ -9,6 +9,15 @@ workflow run_wf {
 
   main:
     output_ch = input_ch
+      // Extract the ID from the input.
+      // If the input is a tarball, strip the suffix.
+      | map{ id, state ->
+        def id_with_suffix = state.input.getFileName().toString()
+        [
+          id,
+          state + [ run_id: id_with_suffix.replaceAll(".tgz", "").replaceAll(".tar.gz", "").replaceAll(".tar", "") ]
```
Comment (DriesSchaumont) [inline src/runner/main.nf:18]:
```suggestion
          state + [ run_id: id_with_suffix - ~/\.(tar.gz|tgz|tar)$/ ]
```

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]
Small syntactic suggestion

File: src/runner/main.nf (hunk 3 above)
Comment (DriesSchaumont) [reply src/runner/main.nf:18]: This only strips trailing occurances.

Comment (tverbeiren) [reply src/runner/main.nf:18]: Ok, sure. Makes sense!

**Pull request merged**